cclib/io/ccio.py

```diff
@@ -161,24 +141,35 @@ def ccread(source, *args, **kwargs):
     Returns:
         a ccData object containing cclib data attributes
     """
-
-    log = ccopen(source, *args, **kwargs)
-    if log:
-        if kwargs.get("verbose", None):
-            print(f"Identified logfile to be in {log.logname} format")
-        # If the input file is a CJSON file and not a standard compchemlog file
-        cjson_as_input = kwargs.get("cjson", False)
-        if cjson_as_input:
-            return log.read_cjson()
+    log = None
+    try:
+        log = ccopen(source, *args, **kwargs)
+        if log:
+            if kwargs.get("verbose", None):
+                print(f"Identified logfile to be in {log.logname} format")
+            # If the input file is a CJSON file and not a standard compchemlog file
+            cjson_as_input = kwargs.get("cjson", False)
+            if cjson_as_input:
+                return log.read_cjson()
+            else:
+                return log.parse()
         else:
-            return log.parse()
-    else:
-        if kwargs.get('verbose', None):
-            print('Attempting to use fallback mechanism to read file')
-        return fallback(source)
-
-
-def ccopen(source, *args, **kwargs):
+            if kwargs.get('verbose', None):
+                print('Attempting to use fallback mechanism to read file')
+            return fallback(source)
+    
+    finally:
+        if log:
+            log.inputfile.close()
+
+
+def ccopen(
+        source: Union[str, typing.IO, FileWrapper, list],
```

Comment:
`list` of what?  It should be `typing.List[...]` and `...` might be the union of the other three (I don't know).